File: Data Structures & Algorithms/validate-parentheses/submission-4.java
class Solution {
    public boolean isValid(String s) {
        Map<Character, Character> mp = new HashMap<>();
        mp.put(')','(');
        mp.put('}','{');
        mp.put(']','[');
        Stack<Character> st = new Stack<>();
        for(char c:s.toCharArray()){
            if(mp.containsKey(c) && !st.isEmpty() && st.peek()==mp.get(c)){
                st.pop();
                continue;
            }
            st.add(c);
        }
        return st.size()==0;
    }
}

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-4.cs
public class Solution {
    public int EvalRPN(string[] tokens) {
        var stack = new Stack<string>();
        foreach(string token in  tokens){
            if(token.Equals("+")||token.Equals("-")||token.Equals("*")||token.Equals("/")){
                var op1 = Int32.Parse(stack.Pop());
                var op2 = Int32.Parse(stack.Pop());
                var res = "";
                if(token.Equals("+")) res = (op1+op2).ToString();
                else if(token.Equals("-")) res = (op2-op1).ToString();
                else if(token.Equals("*")) res = (op1*op2).ToString();
                else if(token.Equals("/")) res = (op2/op1).ToString();
                stack.Push(res.ToString());
            }
            else{
                stack.Push(token);
            }
        }
        return Int32.Parse(stack.Peek());
    }
}
